### PR TITLE
Update SRW for spack-stack 1.5.0 for the cloud platforms

### PR DESCRIPTION
**Pull request by natalie-perlin**
## DESCRIPTION OF CHANGES: 

Updated modulefiles to for running SRW with spack-stack 1.5.0 on the cloud platforms.
Fundamental tests ran successfully, except for the nco test. NCO test produces errors, as described in the comment to this PR. 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [ ] hera.intel
- [ ] orion.intel
- [ ] hercules.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] derecho.intel
- [ ] gaea.intel
- [ ] gaeac5.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [x] NOAA Cloud (indicate which platform)
    - [ ] AWS - still issues with loading a modulefile during task execution
    - [x] GCP
    - [x] Azure
- [ ] Jenkins
- [x] fundamental test suite, except for the NCO test
- [ ] comprehensive tests (specify *which* if a subset was used)

## CHECKLIST

- [ ] My code follows the style guidelines in the Contributor's Guide
- [ ] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [x] help wanted to diagnose NCO test errors

### CODE REVIEW

**Comment by natalie-perlin** [comment]
Running the nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_GFS_v16 , which is one of the new fundamental tests suite, produces errors during workflow generation. The following submitting command was used:
`./run_WE2E_tests.py -m=noaacloud -a cg-noaa -t nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_GFS_v16 -l python -q 2>&1 | tee run_nco_test.log` 

Terminal output with the error message below:

```
Checking that all tests are valid
Will run 1 tests:
/lustre/SRW/srw-dev/tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_GFS_v16.yaml
Calling workflow generation function for test nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_GFS_v16


GRID_DIR will point to a location containing pre-generated files.
Setting GRID_DIR = RRFS_CONUS_25km


*********************************************************************
FATAL ERROR:
Experiment generation failed. See the error message(s) printed below.
For more detailed information, check the log file from the workflow
generation script: log.run_WE2E_tests
*********************************************************************

Traceback (most recent call last):
  File "/lustre/SRW/srw-dev/tests/WE2E/./run_WE2E_tests.py", line 559, in <module>
    run_we2e_tests(homedir,args)
  File "/lustre/SRW/srw-dev/tests/WE2E/./run_WE2E_tests.py", line 241, in run_we2e_tests
    expt_dir = generate_FV3LAM_wflow(ushdir,logfile=f"{ushdir}/log.generate_FV3LAM_wflow",
  File "/lustre/SRW/srw-dev/tests/WE2E/../../ush/generate_FV3LAM_wflow.py", line 72, in generate_FV3LAM_wflow
    expt_config = setup(ushdir,debug=debug)
  File "/lustre/SRW/srw-dev/tests/WE2E/../../ush/setup.py", line 1421, in setup
    raise FileNotFoundError(msg)
FileNotFoundError: 
File directory does not exist!
GRID_DIR needs RRFS_CONUS_25km

**Comment by EdwardSnyder-NOAA** [comment]
That error of not finding a `RRFS_CONUS_25km` is caused be a missing env variable in the noaacloud machine file. Please add this to the noaacloud machine file: `DOMAIN_PREGEN_BASEDIR: /contrib/EPIC/UFS_SRW_data/v2p2/FV3LAM_pregen`. I added this var in for the [release branch](https://github.com/ufs-community/ufs-srweather-app/pull/937/files#diff-d802f6f2ef576e2afdfd92edfe40c305a30d8aad1a8b0d08ef3c6b32fffd983b), but forgot to make the change to the dev branch.

**Comment by natalie-perlin** [comment]
@EdwardSnyder-NOAA - thank you, this indeed helped to fix the issue.
The nco test ran successfully (showing a rocotostat log from GCP):
```
(workflow_tools) [Natalie.Perlin@noaa-gcp:/lustre/SRW/expt_dirs/nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_GFS_v16]$ rocotostat -w FV3LAM_wflow.xml -d FV3LAM_wflow.db -v 10
/apps/rocoto/1.3.3/lib/workflowmgr/launchserver.rb:40: warning: Insecure world writable dir /contrib in PATH, mode 040777
/apps/rocoto/1.3.3/lib/workflowmgr/launchserver.rb:40: warning: Insecure world writable dir /contrib in PATH, mode 040777
       CYCLE                    TASK                       JOBID               STATE         EXIT STATUS     TRIES      DURATION
================================================================================================================================
202208101200           get_extrn_ics                         430           SUCCEEDED                   0         1         194.0
202208101200          get_extrn_lbcs                         431           SUCCEEDED                   0         1         286.0
202208101200         make_ics_mem000                         432           SUCCEEDED                   0         1         109.0
202208101200        make_lbcs_mem000                         433           SUCCEEDED                   0         1         179.0
202208101200         run_fcst_mem000                         434           SUCCEEDED                   0         1         586.0
202208101200    run_post_mem000_f000                         435           SUCCEEDED                   0         1          11.0
202208101200    run_post_mem000_f001                         436           SUCCEEDED                   0         1           9.0
202208101200    run_post_mem000_f002                         437           SUCCEEDED                   0         1          10.0
202208101200    run_post_mem000_f003                         438           SUCCEEDED                   0         1           9.0
202208101200    run_post_mem000_f004                         439           SUCCEEDED                   0         1           9.0
202208101200    run_post_mem000_f005                         440           SUCCEEDED                   0         1          10.0
202208101200    run_post_mem000_f006                         441           SUCCEEDED                   0         1          10.0
(workflow_tools) [Natalie.Perlin@noaa-gcp:/lustre/SRW/expt_dirs/nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_GFS_v16]$
```

File: modulefiles/srw_common.lua [hunk 1]
```diff
@@ -1,10 +1,10 @@
-load_any("jasper/2.0.25","jasper/2.0.32")
-load_any("zlib/1.2.11","zlib/1.2.13")
+load_any("jasper")
```
Comment (RatkoVasic-NOAA) [inline modulefiles/srw_common.lua]: @natalie-perlin  could you please check if we can keep particular library numbers

Comment (natalie-perlin) [reply modulefiles/srw_common.lua]: @RatkoVasic-NOAA - reverting the changes. I've ran it both ways, and both ways work on GCP and Azure. AWS is still an issue in my runs, regardless of the srw_common.lua; the issue could be with my bash environment.

**Comment by natalie-perlin** [comment]
Still getting errors while running tasks on AWS, complaining about loading the modulefile srw_common.lua. The same modulefile is loaded by hand with no issues. 
Log file make_grid_2021051212.log from the experiment grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16 :

```
Lmod has detected the following error: Unable to load module because of error
when evaluating modulefile:
     /lustre/SRW/srw-dev/modulefiles/srw_common.lua: [string
"load_any("jasper/2.0.25","jasper/2.0.32")..."]:1: attempt to call global
'load_any' (a nil value)
     Please check the modulefile and especially if there is a the line number
specified in the above message
While processing the following module(s):
    Module fullname        Module Filename
    ---------------        ---------------
    srw_common             /lustre/SRW/srw-dev/modulefiles/srw_common.lua
    build_noaacloud_intel  /lustre/SRW/srw-dev/modulefiles/build_noaacloud_intel.lua

FATAL ERROR:
ERROR:
  From script:  "load_modules_run_task.sh"
  Full path to script:  "/lustre/SRW/srw-dev/ush/load_modules_run_task.sh"
  Loading of platform- and compiler-specific module file (BUILD_MOD_FN)
for the workflow task specified by task_name failed:
  task_name = "make_grid"
  BUILD_MOD_FN = "build_noaacloud_intel"
Exiting with nonzero status.
End load_modules_run_task.sh at Thu Nov 16 01:51:11 UTC 2023 with error code 1 (time elapsed: 00:00:01)
```

**Comment by RatkoVasic-NOAA** [comment]
To find out what's going on, I suggest you comment every line in srw_common.lua, then see if it fails in build_noaacloud_intel.luaThen, uncomment line by line in srw_common

File: modulefiles/tasks/noaacloud/plot_allvars.local.lua [hunk 2]
```diff
@@ -1 +1,5 @@
-prepend_path("PATH", "/contrib/EPIC/miniconda3/4.12.0/envs/regional_workflow/bin")
+unload("python")
+append_path("MODULEPATH","/contrib/EPIC/miniconda3/modulefiles")
+load(pathJoin("miniconda3", os.getenv("miniconda3_ver") or "4.12.0"))
+
+setenv("SRW_ENV", "regional_workflow")
```
Comment (EdwardSnyder-NOAA) [inline modulefiles/tasks/noaacloud/plot_allvars.local.lua:5]: regional_workflow is needed. please disregard the previous message.

**Comment by EdwardSnyder-NOAA** [comment]
Fundamental suite passed on all three platforms using the changes here except for the changes in the `srw_common.lua` file. Expt dir on all three platforms can be found here: /contrib/Edward.Snyder/PR_ss150/expt_dirs
AWS:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE              59.08
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              64.04
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2        COMPLETE              15.29
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              81.03
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR          COMPLETE             299.12
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0              COMPLETE              77.29
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16                COMPLETE             132.44
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             728.29
```
Azure:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE              24.31
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              27.81
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2        COMPLETE              15.76
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              38.29
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR          COMPLETE              59.99
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0              COMPLETE              31.85
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16                COMPLETE              52.19
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             250.20
```
GCP:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE              59.08
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              64.04
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2        COMPLETE              15.29
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              81.03
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR          COMPLETE              25.52
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0              COMPLETE              77.29
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16                COMPLETE             132.44
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             454.42
```

**Review by EdwardSnyder-NOAA — APPROVED** [review]

**Review by RatkoVasic-NOAA — APPROVED** [review]

**Pull request merged**